### PR TITLE
[PATCH v4] validation: dma: fix max segment length

### PR DESCRIPTION
**Pull request by PavanNikhilesh**
Max segment length should be calculated based on the DMA device
capabilities such as max supported segments and segment length.

Fixes: 1aaf5cf55a0c ("validation: dma: add tests for maximum segment amount transfers")

Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Reviewed-by: Tuomas Taipale <tuomas.taipale@nokia.com>

### CODE REVIEW

File: test/validation/api/dma/dma.c [hunk 1]
```diff
@@ -1320,7 +1320,7 @@ static void get_seg_lens(uint32_t max_len, uint32_t *src, uint32_t *dst)
 {
 	uint32_t src_segs = *src, dst_segs = *dst;
 
-	*src = max_len / src_segs;
+	*src = MIN(max_len / src_segs, global.dma_capa.max_seg_len);
 	*dst = *src * src_segs / dst_segs + *src * src_segs % dst_segs;
```
Comment (TuomasTaipale) [inline test/validation/api/dma/dma.c:1324]: Good catch, though I believe this might cause failures in case `max_src_segs` and `max_dst_segs` capabilities differ as the total data amount between source and destination segments could end up being different (if destination length was larger than maximum and needed truncation as well). Untested, but this could work:
```diff
@@ -1318,8 +1318,9 @@ static void test_dma_addr_to_addr_sync_res(void)
 
 static void get_seg_lens(uint32_t max_len, uint32_t *src, uint32_t *dst)
 {
-       uint32_t src_segs = *src, dst_segs = *dst;
+       uint32_t src_segs = *src, dst_segs = *dst, denom = MIN(src_segs, dst_segs);
 
+       max_len = MIN(max_len / denom, global.dma_capa.max_seg_len) * denom;
        *src = max_len / src_segs;
        *dst = *src * src_segs / dst_segs + *src * src_segs % dst_segs;
 }
```

Comment (PavanNikhilesh) [reply test/validation/api/dma/dma.c:1324]: Thanks, I will spin up the next version with the above changes.

**Review by TuomasTaipale — APPROVED** [review]
Reviewed-by: Tuomas Taipale \<tuomas.taipale@nokia.com\>

**Comment by TuomasTaipale** [comment]
@PavanNikhilesh please add the reviewed-by tag to the commit message so we can proceed to merging this.
